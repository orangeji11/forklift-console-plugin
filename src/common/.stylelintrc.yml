extends: "../eslint-plugin/stylelintrc.yml"
