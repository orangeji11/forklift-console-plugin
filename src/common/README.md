# @kubev2v/common

Shared components and urilities that do not relay on forklift plugin contexts or enviorment.