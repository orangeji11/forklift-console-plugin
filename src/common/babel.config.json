{
  "sourceType": "unambiguous",
  "presets": [
    [
      "@babel/preset-env",
      {
        "targets": {
          "chrome": 100
        }
      }
    ],
    "@babel/preset-react",
    [
      "@babel/preset-typescript",
      {
          "onlyRemoveTypeImports": true,
          "allExtensions": true,
          "isTSX": true
      }
    ]
  ],
  "plugins": []
}