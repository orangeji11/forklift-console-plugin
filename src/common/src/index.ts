// @index(['./*', /__/g], f => `export * from '${f.path}';`)
export * from './components';
export * from './hooks';
export * from './utils';
// @endindex
