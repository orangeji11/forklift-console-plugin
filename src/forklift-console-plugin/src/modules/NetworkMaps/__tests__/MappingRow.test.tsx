import * as React from 'react';
import { MemoryRouter } from 'react-router';

import { withQueryClient } from '@kubev2v/common';
import { cleanup, render } from '@testing-library/react';

import { FlatNetworkMapping } from '../dataForNetwork';
import NetworkMappingRow from '../NetworkMappingRow';
import { fieldsMetadataFactory as networkFieldsFactory } from '../NetworkMappingsPage';

import MERGED_NETWORK_DATA from './mergedNetworkData.json';

// Mock translation function.
const t = (s) => s;
// Create a field metadata
const networkFields = networkFieldsFactory(t);

afterEach(cleanup);

describe('NetworkMap rows', () => {
  const netTuples = MERGED_NETWORK_DATA.map((net) => [net.name, net]);
  test.each(netTuples)('%s', (description, net) => {
    const Wrapped = withQueryClient(() => (
      <MemoryRouter>
        <table>
          <tbody>
            <NetworkMappingRow
              resourceFields={networkFields.filter((f) => !f.isHidden)}
              namespace={undefined}
              resourceData={net as FlatNetworkMapping}
              resourceIndex={0}
            />
          </tbody>
        </table>
      </MemoryRouter>
    ));
    const { asFragment } = render(<Wrapped />);
    expect(asFragment()).toMatchSnapshot();
  });
});
