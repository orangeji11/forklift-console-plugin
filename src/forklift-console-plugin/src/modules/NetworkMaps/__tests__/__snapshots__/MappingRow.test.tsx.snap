// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NetworkMap rows plantest1-generated-zxcv 1`] = `
<DocumentFragment>
  <table>
    <tbody>
      <tr
        class=""
        data-ouia-component-id="OUIA-Generated-TableRow-3"
        data-ouia-component-type="PF4/TableRow"
        data-ouia-safe="true"
      >
        <td
          class="pf-c-table__toggle"
        >
          <button
            aria-disabled="false"
            aria-expanded="false"
            aria-label="Details"
            aria-labelledby="simple-node0 expand-toggle0"
            class="pf-c-button pf-m-plain"
            data-ouia-component-id="OUIA-Generated-Button-plain-2"
            data-ouia-component-type="PF4/Button"
            data-ouia-safe="true"
            id="expand-toggle0"
            type="button"
          >
            <div
              class="pf-c-table__toggle-icon"
            >
              <svg
                aria-hidden="true"
                fill="currentColor"
                height="1em"
                role="img"
                style="vertical-align: -0.125em;"
                viewBox="0 0 320 512"
                width="1em"
              >
                <path
                  d="M143 352.3L7 216.3c-9.4-9.4-9.4-24.6 0-33.9l22.6-22.6c9.4-9.4 24.6-9.4 33.9 0l96.4 96.4 96.4-96.4c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9l-136 136c-9.2 9.4-24.4 9.4-33.8 0z"
                />
              </svg>
            </div>
          </button>
        </td>
        <td
          class=""
          data-label="Name"
        >
          <span
            class="forklift-table__flex-cell"
          >
            <div
              class="ResourceLink_mock"
            >
              name: plantest1-generated-zxcv, gvk: forklift.konveyor.io~v1beta1~NetworkMap, ns: konveyor-forklift
            </div>
            <span
              class="pf-c-label pf-m-compact forklift-table__flex-cell-label"
            >
              <span
                class="pf-c-label__content"
              >
                managed
              </span>
            </span>
          </span>
        </td>
        <td
          class=""
          data-label="Namespace"
        >
          <div
            class="ResourceLink_mock"
          >
            name: konveyor-forklift, gvk: ~~, ns: undefined
          </div>
        </td>
        <td
          class=""
          data-label="Source provider"
        >
          <div
            class="ResourceLink_mock"
          >
            name: vcenter-1, gvk: forklift.konveyor.io~v1beta1~Provider, ns: konveyor-forklift
          </div>
        </td>
        <td
          class=""
          data-label="Target provider"
        >
          <div
            class="ResourceLink_mock"
          >
            name: ocpv-1, gvk: forklift.konveyor.io~v1beta1~Provider, ns: konveyor-forklift
          </div>
        </td>
        <td
          class="pf-c-table__compound-expansion-toggle"
          data-label="From"
        >
          <button
            aria-expanded="false"
            class="pf-c-table__button"
            type="button"
          >
            <span
              class="pf-c-table__text"
            >
              <svg
                aria-hidden="true"
                fill="currentColor"
                height="1em"
                role="img"
                style="vertical-align: -0.125em;"
                viewBox="0 0 1088 1024"
                width="1em"
              >
                <path
                  d="M574,320 L574,448 L896,448 C931.332514,448.033078 959.966922,476.667486 960,512 L960,512 L960,640 L1024,640 C1059.33251,640.033078 1087.96692,668.667486 1088,704 L1088,704 L1088,896 C1087.96692,931.332514 1059.33251,959.966922 1024,960 L1024,960 L832,960 C796.667486,959.966922 768.033078,931.332514 768,896 L768,896 L768,704 C768.033078,668.667486 796.667486,640.033078 832,640 L832,640 L896,640 L896,512 L574,512 L574,640 L639,640 C674.332514,640.033078 702.966922,668.667486 703,704 L703,704 L703,896 C702.966922,931.332514 674.332514,959.966922 639,960 L639,960 L447,960 C411.667486,959.966922 383.033078,931.332514 383,896 L383,896 L383,704 C383.033078,668.667486 411.667486,640.033078 447,640 L447,640 L510,640 L510,512 L192,512 L192,640 L256,640 C291.332514,640.033078 319.966922,668.667486 320,704 L320,704 L320,896 C319.966922,931.332514 291.332514,959.966922 256,960 L256,960 L64,960 C28.6674863,959.966922 0.0330777378,931.332514 0,896 L0,896 L0,704 C0.0330777378,668.667486 28.6674863,640.033078 64,640 L64,640 L128,640 L128,512 C128.033078,476.667486 156.667486,448.033078 192,448 L192,448 L510,448 L510,320 L574,320 Z M1024,64 L1024,256 L64,256 L64,64 L1024,64 Z M704,192 L960.5,192 L960.5,128 L704,128 L704,192 Z"
                />
              </svg>
               1
            </span>
          </button>
        </td>
        <td
          class=""
          data-label="Status"
        >
          <button
            aria-disabled="false"
            aria-label="Ready"
            class="pf-c-button pf-m-link pf-m-inline"
            data-ouia-component-id="OUIA-Generated-Button-link-2"
            data-ouia-component-type="PF4/Button"
            data-ouia-safe="true"
            type="button"
          >
            <div
              class="pf-l-flex pf-m-space-items-xs pf-m-inline-flex pf-m-nowrap"
            >
              <div
                class=""
              >
                <div
                  data-test-element-name="GreenCheckCircleIcon"
                />
              </div>
              <div
                class=""
              >
                Ready
              </div>
            </div>
          </button>
        </td>
        <td
          class="pf-c-table__compound-expansion-toggle"
          data-label="To"
        >
          <button
            aria-expanded="false"
            class="pf-c-table__button"
            type="button"
          >
            <span
              class="pf-c-table__text"
            >
              <span
                class="forklift-table__flex-labels-with-gaps"
              >
                <span
                  class="pf-c-label pf-m-blue"
                >
                  <span
                    class="pf-c-label__content"
                  >
                    openshift-migration/ocp-network-1
                  </span>
                </span>
              </span>
            </span>
          </button>
        </td>
        <td
          class=""
          data-label=""
        >
          <div
            data-test-element-name="ActionServiceProvider"
          />
        </td>
      </tr>
      <tr
        class="pf-c-table__expandable-row"
        data-ouia-component-id="OUIA-Generated-TableRow-4"
        data-ouia-component-type="PF4/TableRow"
        data-ouia-safe="true"
        hidden=""
      >
        <td
          class="pf-m-no-padding"
          colspan="8"
          data-label="Mapping graph"
        />
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;

exports[`NetworkMap rows vcenter1-netstore-to-ocp1-network1 1`] = `
<DocumentFragment>
  <table>
    <tbody>
      <tr
        class=""
        data-ouia-component-id="OUIA-Generated-TableRow-1"
        data-ouia-component-type="PF4/TableRow"
        data-ouia-safe="true"
      >
        <td
          class="pf-c-table__toggle"
        >
          <button
            aria-disabled="false"
            aria-expanded="false"
            aria-label="Details"
            aria-labelledby="simple-node0 expand-toggle0"
            class="pf-c-button pf-m-plain"
            data-ouia-component-id="OUIA-Generated-Button-plain-1"
            data-ouia-component-type="PF4/Button"
            data-ouia-safe="true"
            id="expand-toggle0"
            type="button"
          >
            <div
              class="pf-c-table__toggle-icon"
            >
              <svg
                aria-hidden="true"
                fill="currentColor"
                height="1em"
                role="img"
                style="vertical-align: -0.125em;"
                viewBox="0 0 320 512"
                width="1em"
              >
                <path
                  d="M143 352.3L7 216.3c-9.4-9.4-9.4-24.6 0-33.9l22.6-22.6c9.4-9.4 24.6-9.4 33.9 0l96.4 96.4 96.4-96.4c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9l-136 136c-9.2 9.4-24.4 9.4-33.8 0z"
                />
              </svg>
            </div>
          </button>
        </td>
        <td
          class=""
          data-label="Name"
        >
          <span
            class="forklift-table__flex-cell"
          >
            <div
              class="ResourceLink_mock"
            >
              name: vcenter1-netstore-to-ocp1-network1, gvk: forklift.konveyor.io~v1beta1~NetworkMap, ns: konveyor-forklift
            </div>
          </span>
        </td>
        <td
          class=""
          data-label="Namespace"
        >
          <div
            class="ResourceLink_mock"
          >
            name: konveyor-forklift, gvk: ~~, ns: undefined
          </div>
        </td>
        <td
          class=""
          data-label="Source provider"
        >
          <div
            class="ResourceLink_mock"
          >
            name: vcenter-1, gvk: forklift.konveyor.io~v1beta1~Provider, ns: konveyor-forklift
          </div>
        </td>
        <td
          class=""
          data-label="Target provider"
        >
          <div
            class="ResourceLink_mock"
          >
            name: ocpv-1, gvk: forklift.konveyor.io~v1beta1~Provider, ns: konveyor-forklift
          </div>
        </td>
        <td
          class="pf-c-table__compound-expansion-toggle"
          data-label="From"
        >
          <button
            aria-expanded="false"
            class="pf-c-table__button"
            type="button"
          >
            <span
              class="pf-c-table__text"
            >
              <svg
                aria-hidden="true"
                fill="currentColor"
                height="1em"
                role="img"
                style="vertical-align: -0.125em;"
                viewBox="0 0 1088 1024"
                width="1em"
              >
                <path
                  d="M574,320 L574,448 L896,448 C931.332514,448.033078 959.966922,476.667486 960,512 L960,512 L960,640 L1024,640 C1059.33251,640.033078 1087.96692,668.667486 1088,704 L1088,704 L1088,896 C1087.96692,931.332514 1059.33251,959.966922 1024,960 L1024,960 L832,960 C796.667486,959.966922 768.033078,931.332514 768,896 L768,896 L768,704 C768.033078,668.667486 796.667486,640.033078 832,640 L832,640 L896,640 L896,512 L574,512 L574,640 L639,640 C674.332514,640.033078 702.966922,668.667486 703,704 L703,704 L703,896 C702.966922,931.332514 674.332514,959.966922 639,960 L639,960 L447,960 C411.667486,959.966922 383.033078,931.332514 383,896 L383,896 L383,704 C383.033078,668.667486 411.667486,640.033078 447,640 L447,640 L510,640 L510,512 L192,512 L192,640 L256,640 C291.332514,640.033078 319.966922,668.667486 320,704 L320,704 L320,896 C319.966922,931.332514 291.332514,959.966922 256,960 L256,960 L64,960 C28.6674863,959.966922 0.0330777378,931.332514 0,896 L0,896 L0,704 C0.0330777378,668.667486 28.6674863,640.033078 64,640 L64,640 L128,640 L128,512 C128.033078,476.667486 156.667486,448.033078 192,448 L192,448 L510,448 L510,320 L574,320 Z M1024,64 L1024,256 L64,256 L64,64 L1024,64 Z M704,192 L960.5,192 L960.5,128 L704,128 L704,192 Z"
                />
              </svg>
               1
            </span>
          </button>
        </td>
        <td
          class=""
          data-label="Status"
        >
          <button
            aria-disabled="false"
            aria-label="Ready"
            class="pf-c-button pf-m-link pf-m-inline"
            data-ouia-component-id="OUIA-Generated-Button-link-1"
            data-ouia-component-type="PF4/Button"
            data-ouia-safe="true"
            type="button"
          >
            <div
              class="pf-l-flex pf-m-space-items-xs pf-m-inline-flex pf-m-nowrap"
            >
              <div
                class=""
              >
                <div
                  data-test-element-name="GreenCheckCircleIcon"
                />
              </div>
              <div
                class=""
              >
                Ready
              </div>
            </div>
          </button>
        </td>
        <td
          class="pf-c-table__compound-expansion-toggle"
          data-label="To"
        >
          <button
            aria-expanded="false"
            class="pf-c-table__button"
            type="button"
          >
            <span
              class="pf-c-table__text"
            >
              <span
                class="forklift-table__flex-labels-with-gaps"
              >
                <span
                  class="pf-c-label pf-m-blue"
                >
                  <span
                    class="pf-c-label__content"
                  >
                    openshift-migration/ocp-network-1
                  </span>
                </span>
              </span>
            </span>
          </button>
        </td>
        <td
          class=""
          data-label=""
        >
          <div
            data-test-element-name="ActionServiceProvider"
          />
        </td>
      </tr>
      <tr
        class="pf-c-table__expandable-row"
        data-ouia-component-id="OUIA-Generated-TableRow-2"
        data-ouia-component-type="PF4/TableRow"
        data-ouia-safe="true"
        hidden=""
      >
        <td
          class="pf-m-no-padding"
          colspan="8"
          data-label="Mapping graph"
        />
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;

exports[`NetworkMap rows vcenter1-netstore-to-ocp1-network2 1`] = `
<DocumentFragment>
  <table>
    <tbody>
      <tr
        class=""
        data-ouia-component-id="OUIA-Generated-TableRow-5"
        data-ouia-component-type="PF4/TableRow"
        data-ouia-safe="true"
      >
        <td
          class="pf-c-table__toggle"
        >
          <button
            aria-disabled="false"
            aria-expanded="false"
            aria-label="Details"
            aria-labelledby="simple-node0 expand-toggle0"
            class="pf-c-button pf-m-plain"
            data-ouia-component-id="OUIA-Generated-Button-plain-3"
            data-ouia-component-type="PF4/Button"
            data-ouia-safe="true"
            id="expand-toggle0"
            type="button"
          >
            <div
              class="pf-c-table__toggle-icon"
            >
              <svg
                aria-hidden="true"
                fill="currentColor"
                height="1em"
                role="img"
                style="vertical-align: -0.125em;"
                viewBox="0 0 320 512"
                width="1em"
              >
                <path
                  d="M143 352.3L7 216.3c-9.4-9.4-9.4-24.6 0-33.9l22.6-22.6c9.4-9.4 24.6-9.4 33.9 0l96.4 96.4 96.4-96.4c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9l-136 136c-9.2 9.4-24.4 9.4-33.8 0z"
                />
              </svg>
            </div>
          </button>
        </td>
        <td
          class=""
          data-label="Name"
        >
          <span
            class="forklift-table__flex-cell"
          >
            <div
              class="ResourceLink_mock"
            >
              name: vcenter1-netstore-to-ocp1-network2, gvk: forklift.konveyor.io~v1beta1~NetworkMap, ns: konveyor-forklift
            </div>
          </span>
        </td>
        <td
          class=""
          data-label="Namespace"
        >
          <div
            class="ResourceLink_mock"
          >
            name: konveyor-forklift, gvk: ~~, ns: undefined
          </div>
        </td>
        <td
          class=""
          data-label="Source provider"
        >
          <div
            class="ResourceLink_mock"
          >
            name: vcenter-1, gvk: forklift.konveyor.io~v1beta1~Provider, ns: konveyor-forklift
          </div>
        </td>
        <td
          class=""
          data-label="Target provider"
        >
          <div
            class="ResourceLink_mock"
          >
            name: ocpv-2, gvk: forklift.konveyor.io~v1beta1~Provider, ns: konveyor-forklift
          </div>
        </td>
        <td
          class="pf-c-table__compound-expansion-toggle"
          data-label="From"
        >
          <button
            aria-expanded="false"
            class="pf-c-table__button"
            type="button"
          >
            <span
              class="pf-c-table__text"
            >
              <svg
                aria-hidden="true"
                fill="currentColor"
                height="1em"
                role="img"
                style="vertical-align: -0.125em;"
                viewBox="0 0 1088 1024"
                width="1em"
              >
                <path
                  d="M574,320 L574,448 L896,448 C931.332514,448.033078 959.966922,476.667486 960,512 L960,512 L960,640 L1024,640 C1059.33251,640.033078 1087.96692,668.667486 1088,704 L1088,704 L1088,896 C1087.96692,931.332514 1059.33251,959.966922 1024,960 L1024,960 L832,960 C796.667486,959.966922 768.033078,931.332514 768,896 L768,896 L768,704 C768.033078,668.667486 796.667486,640.033078 832,640 L832,640 L896,640 L896,512 L574,512 L574,640 L639,640 C674.332514,640.033078 702.966922,668.667486 703,704 L703,704 L703,896 C702.966922,931.332514 674.332514,959.966922 639,960 L639,960 L447,960 C411.667486,959.966922 383.033078,931.332514 383,896 L383,896 L383,704 C383.033078,668.667486 411.667486,640.033078 447,640 L447,640 L510,640 L510,512 L192,512 L192,640 L256,640 C291.332514,640.033078 319.966922,668.667486 320,704 L320,704 L320,896 C319.966922,931.332514 291.332514,959.966922 256,960 L256,960 L64,960 C28.6674863,959.966922 0.0330777378,931.332514 0,896 L0,896 L0,704 C0.0330777378,668.667486 28.6674863,640.033078 64,640 L64,640 L128,640 L128,512 C128.033078,476.667486 156.667486,448.033078 192,448 L192,448 L510,448 L510,320 L574,320 Z M1024,64 L1024,256 L64,256 L64,64 L1024,64 Z M704,192 L960.5,192 L960.5,128 L704,128 L704,192 Z"
                />
              </svg>
               {{groupCount}} Groups
            </span>
          </button>
        </td>
        <td
          class=""
          data-label="Status"
        >
          <button
            aria-disabled="false"
            aria-label="Ready"
            class="pf-c-button pf-m-link pf-m-inline"
            data-ouia-component-id="OUIA-Generated-Button-link-3"
            data-ouia-component-type="PF4/Button"
            data-ouia-safe="true"
            type="button"
          >
            <div
              class="pf-l-flex pf-m-space-items-xs pf-m-inline-flex pf-m-nowrap"
            >
              <div
                class=""
              >
                <div
                  data-test-element-name="GreenCheckCircleIcon"
                />
              </div>
              <div
                class=""
              >
                Ready
              </div>
            </div>
          </button>
        </td>
        <td
          class="pf-c-table__compound-expansion-toggle"
          data-label="To"
        >
          <button
            aria-expanded="false"
            class="pf-c-table__button"
            type="button"
          >
            <span
              class="pf-c-table__text"
            >
              <span
                class="forklift-table__flex-labels-with-gaps"
              >
                <span
                  class="pf-c-label pf-m-blue"
                >
                  <span
                    class="pf-c-label__content"
                  >
                    openshift-migration/ocp-network-2
                  </span>
                </span>
                <span
                  class="pf-c-label pf-m-blue"
                >
                  <span
                    class="pf-c-label__content"
                  >
                    openshift-migration/ocp-network-1
                  </span>
                </span>
                <span
                  class="pf-c-label pf-m-blue"
                >
                  <span
                    class="pf-c-label__content"
                  >
                    openshift-migration/ocp-network-3
                  </span>
                </span>
              </span>
            </span>
          </button>
        </td>
        <td
          class=""
          data-label=""
        >
          <div
            data-test-element-name="ActionServiceProvider"
          />
        </td>
      </tr>
      <tr
        class="pf-c-table__expandable-row"
        data-ouia-component-id="OUIA-Generated-TableRow-6"
        data-ouia-component-type="PF4/TableRow"
        data-ouia-safe="true"
        hidden=""
      >
        <td
          class="pf-m-no-padding"
          colspan="8"
          data-label="Mapping graph"
        />
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;

exports[`NetworkMap rows vcenter3-invalid-network-map 1`] = `
<DocumentFragment>
  <table>
    <tbody>
      <tr
        class=""
        data-ouia-component-id="OUIA-Generated-TableRow-7"
        data-ouia-component-type="PF4/TableRow"
        data-ouia-safe="true"
      >
        <td
          class="pf-c-table__toggle"
        >
          <button
            aria-disabled="false"
            aria-expanded="false"
            aria-label="Details"
            aria-labelledby="simple-node0 expand-toggle0"
            class="pf-c-button pf-m-plain"
            data-ouia-component-id="OUIA-Generated-Button-plain-4"
            data-ouia-component-type="PF4/Button"
            data-ouia-safe="true"
            id="expand-toggle0"
            type="button"
          >
            <div
              class="pf-c-table__toggle-icon"
            >
              <svg
                aria-hidden="true"
                fill="currentColor"
                height="1em"
                role="img"
                style="vertical-align: -0.125em;"
                viewBox="0 0 320 512"
                width="1em"
              >
                <path
                  d="M143 352.3L7 216.3c-9.4-9.4-9.4-24.6 0-33.9l22.6-22.6c9.4-9.4 24.6-9.4 33.9 0l96.4 96.4 96.4-96.4c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9l-136 136c-9.2 9.4-24.4 9.4-33.8 0z"
                />
              </svg>
            </div>
          </button>
        </td>
        <td
          class=""
          data-label="Name"
        >
          <span
            class="forklift-table__flex-cell"
          >
            <div
              class="ResourceLink_mock"
            >
              name: vcenter3-invalid-network-map, gvk: forklift.konveyor.io~v1beta1~NetworkMap, ns: konveyor-forklift
            </div>
          </span>
        </td>
        <td
          class=""
          data-label="Namespace"
        >
          <div
            class="ResourceLink_mock"
          >
            name: konveyor-forklift, gvk: ~~, ns: undefined
          </div>
        </td>
        <td
          class=""
          data-label="Source provider"
        >
          <div
            class="ResourceLink_mock"
          >
            name: vcenter-1, gvk: forklift.konveyor.io~v1beta1~Provider, ns: konveyor-forklift
          </div>
        </td>
        <td
          class=""
          data-label="Target provider"
        >
          <div
            class="ResourceLink_mock"
          >
            name: ocpv-1, gvk: forklift.konveyor.io~v1beta1~Provider, ns: konveyor-forklift
          </div>
        </td>
        <td
          class="pf-c-table__compound-expansion-toggle"
          data-label="From"
        >
          <button
            aria-expanded="false"
            class="pf-c-table__button"
            type="button"
          >
            <span
              class="pf-c-table__text"
            >
              <svg
                aria-hidden="true"
                fill="currentColor"
                height="1em"
                role="img"
                style="vertical-align: -0.125em;"
                viewBox="0 0 1088 1024"
                width="1em"
              >
                <path
                  d="M574,320 L574,448 L896,448 C931.332514,448.033078 959.966922,476.667486 960,512 L960,512 L960,640 L1024,640 C1059.33251,640.033078 1087.96692,668.667486 1088,704 L1088,704 L1088,896 C1087.96692,931.332514 1059.33251,959.966922 1024,960 L1024,960 L832,960 C796.667486,959.966922 768.033078,931.332514 768,896 L768,896 L768,704 C768.033078,668.667486 796.667486,640.033078 832,640 L832,640 L896,640 L896,512 L574,512 L574,640 L639,640 C674.332514,640.033078 702.966922,668.667486 703,704 L703,704 L703,896 C702.966922,931.332514 674.332514,959.966922 639,960 L639,960 L447,960 C411.667486,959.966922 383.033078,931.332514 383,896 L383,896 L383,704 C383.033078,668.667486 411.667486,640.033078 447,640 L447,640 L510,640 L510,512 L192,512 L192,640 L256,640 C291.332514,640.033078 319.966922,668.667486 320,704 L320,704 L320,896 C319.966922,931.332514 291.332514,959.966922 256,960 L256,960 L64,960 C28.6674863,959.966922 0.0330777378,931.332514 0,896 L0,896 L0,704 C0.0330777378,668.667486 28.6674863,640.033078 64,640 L64,640 L128,640 L128,512 C128.033078,476.667486 156.667486,448.033078 192,448 L192,448 L510,448 L510,320 L574,320 Z M1024,64 L1024,256 L64,256 L64,64 L1024,64 Z M704,192 L960.5,192 L960.5,128 L704,128 L704,192 Z"
                />
              </svg>
               1
            </span>
          </button>
        </td>
        <td
          class=""
          data-label="Status"
        >
          <button
            aria-disabled="false"
            aria-label="Not Ready"
            class="pf-c-button pf-m-link pf-m-inline"
            data-ouia-component-id="OUIA-Generated-Button-link-4"
            data-ouia-component-type="PF4/Button"
            data-ouia-safe="true"
            type="button"
          >
            <div
              class="pf-l-flex pf-m-space-items-xs pf-m-inline-flex pf-m-nowrap"
            >
              <div
                class=""
              >
                <div
                  data-test-element-name="RedExclamationCircleIcon"
                />
              </div>
              <div
                class=""
              >
                Not Ready
              </div>
            </div>
          </button>
        </td>
        <td
          class="pf-c-table__compound-expansion-toggle"
          data-label="To"
        >
          <button
            aria-expanded="false"
            class="pf-c-table__button"
            type="button"
          >
            <span
              class="pf-c-table__text"
            >
              <span
                class="forklift-table__flex-labels-with-gaps"
              >
                <span
                  class="pf-c-label pf-m-blue"
                >
                  <span
                    class="pf-c-label__content"
                  >
                    doesnt-matter/missing-network
                  </span>
                </span>
              </span>
            </span>
          </button>
        </td>
        <td
          class=""
          data-label=""
        >
          <div
            data-test-element-name="ActionServiceProvider"
          />
        </td>
      </tr>
      <tr
        class="pf-c-table__expandable-row"
        data-ouia-component-id="OUIA-Generated-TableRow-8"
        data-ouia-component-type="PF4/TableRow"
        data-ouia-safe="true"
        hidden=""
      >
        <td
          class="pf-m-no-padding"
          colspan="8"
          data-label="Mapping graph"
        />
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;

exports[`NetworkMap rows vcenter4-invalid-network-map 1`] = `
<DocumentFragment>
  <table>
    <tbody>
      <tr
        class=""
        data-ouia-component-id="OUIA-Generated-TableRow-9"
        data-ouia-component-type="PF4/TableRow"
        data-ouia-safe="true"
      >
        <td
          class="pf-c-table__toggle"
        >
          <button
            aria-disabled="false"
            aria-expanded="false"
            aria-label="Details"
            aria-labelledby="simple-node0 expand-toggle0"
            class="pf-c-button pf-m-plain"
            data-ouia-component-id="OUIA-Generated-Button-plain-5"
            data-ouia-component-type="PF4/Button"
            data-ouia-safe="true"
            id="expand-toggle0"
            type="button"
          >
            <div
              class="pf-c-table__toggle-icon"
            >
              <svg
                aria-hidden="true"
                fill="currentColor"
                height="1em"
                role="img"
                style="vertical-align: -0.125em;"
                viewBox="0 0 320 512"
                width="1em"
              >
                <path
                  d="M143 352.3L7 216.3c-9.4-9.4-9.4-24.6 0-33.9l22.6-22.6c9.4-9.4 24.6-9.4 33.9 0l96.4 96.4 96.4-96.4c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9l-136 136c-9.2 9.4-24.4 9.4-33.8 0z"
                />
              </svg>
            </div>
          </button>
        </td>
        <td
          class=""
          data-label="Name"
        >
          <span
            class="forklift-table__flex-cell"
          >
            <div
              class="ResourceLink_mock"
            >
              name: vcenter4-invalid-network-map, gvk: forklift.konveyor.io~v1beta1~NetworkMap, ns: konveyor-forklift
            </div>
          </span>
        </td>
        <td
          class=""
          data-label="Namespace"
        >
          <div
            class="ResourceLink_mock"
          >
            name: konveyor-forklift, gvk: ~~, ns: undefined
          </div>
        </td>
        <td
          class=""
          data-label="Source provider"
        >
          <div
            class="ResourceLink_mock"
          >
            name: unknown-provider, gvk: forklift.konveyor.io~v1beta1~Provider, ns: konveyor-forklift
          </div>
        </td>
        <td
          class=""
          data-label="Target provider"
        >
          <div
            class="ResourceLink_mock"
          >
            name: ocpv-1, gvk: forklift.konveyor.io~v1beta1~Provider, ns: konveyor-forklift
          </div>
        </td>
        <td
          class="pf-c-table__compound-expansion-toggle"
          data-label="From"
        >
          <button
            aria-expanded="false"
            class="pf-c-table__button"
            type="button"
          >
            <span
              class="pf-c-table__text"
            >
              <svg
                aria-hidden="true"
                fill="currentColor"
                height="1em"
                role="img"
                style="vertical-align: -0.125em;"
                viewBox="0 0 1088 1024"
                width="1em"
              >
                <path
                  d="M574,320 L574,448 L896,448 C931.332514,448.033078 959.966922,476.667486 960,512 L960,512 L960,640 L1024,640 C1059.33251,640.033078 1087.96692,668.667486 1088,704 L1088,704 L1088,896 C1087.96692,931.332514 1059.33251,959.966922 1024,960 L1024,960 L832,960 C796.667486,959.966922 768.033078,931.332514 768,896 L768,896 L768,704 C768.033078,668.667486 796.667486,640.033078 832,640 L832,640 L896,640 L896,512 L574,512 L574,640 L639,640 C674.332514,640.033078 702.966922,668.667486 703,704 L703,704 L703,896 C702.966922,931.332514 674.332514,959.966922 639,960 L639,960 L447,960 C411.667486,959.966922 383.033078,931.332514 383,896 L383,896 L383,704 C383.033078,668.667486 411.667486,640.033078 447,640 L447,640 L510,640 L510,512 L192,512 L192,640 L256,640 C291.332514,640.033078 319.966922,668.667486 320,704 L320,704 L320,896 C319.966922,931.332514 291.332514,959.966922 256,960 L256,960 L64,960 C28.6674863,959.966922 0.0330777378,931.332514 0,896 L0,896 L0,704 C0.0330777378,668.667486 28.6674863,640.033078 64,640 L64,640 L128,640 L128,512 C128.033078,476.667486 156.667486,448.033078 192,448 L192,448 L510,448 L510,320 L574,320 Z M1024,64 L1024,256 L64,256 L64,64 L1024,64 Z M704,192 L960.5,192 L960.5,128 L704,128 L704,192 Z"
                />
              </svg>
               1
            </span>
          </button>
        </td>
        <td
          class=""
          data-label="Status"
        >
          <button
            aria-disabled="false"
            aria-label="Not Ready"
            class="pf-c-button pf-m-link pf-m-inline"
            data-ouia-component-id="OUIA-Generated-Button-link-5"
            data-ouia-component-type="PF4/Button"
            data-ouia-safe="true"
            type="button"
          >
            <div
              class="pf-l-flex pf-m-space-items-xs pf-m-inline-flex pf-m-nowrap"
            >
              <div
                class=""
              >
                <div
                  data-test-element-name="RedExclamationCircleIcon"
                />
              </div>
              <div
                class=""
              >
                Not Ready
              </div>
            </div>
          </button>
        </td>
        <td
          class="pf-c-table__compound-expansion-toggle"
          data-label="To"
        >
          <button
            aria-expanded="false"
            class="pf-c-table__button"
            type="button"
          >
            <span
              class="pf-c-table__text"
            >
              <span
                class="forklift-table__flex-labels-with-gaps"
              >
                <span
                  class="pf-c-label pf-m-blue"
                >
                  <span
                    class="pf-c-label__content"
                  >
                    openshift-migration/ocp-network-3
                  </span>
                </span>
              </span>
            </span>
          </button>
        </td>
        <td
          class=""
          data-label=""
        >
          <div
            data-test-element-name="ActionServiceProvider"
          />
        </td>
      </tr>
      <tr
        class="pf-c-table__expandable-row"
        data-ouia-component-id="OUIA-Generated-TableRow-10"
        data-ouia-component-type="PF4/TableRow"
        data-ouia-safe="true"
        hidden=""
      >
        <td
          class="pf-m-no-padding"
          colspan="8"
          data-label="Mapping graph"
        />
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;
